bolt-cli/src/commands/operators.rs

```diff
@@ -140,22 +186,124 @@ impl OperatorsCommand {
+
+                    let deployments = deployments_for_chain(chain);
+
+                    info!(operator = %signer.address(), rpc = %operator_rpc, ?chain, "Registering Symbiotic operator");
+
+                    if !request_confirmation() {
+                        info!("Aborting");
+                        return Ok(());
+                    }
+
+                    // Check if operator is opted in to the bolt network
+                    if !IOptInService::new(
+                        deployments.symbiotic.network_opt_in_service,
+                        provider.clone(),
+                    )
+                    .isOptedIn(signer.address(), deployments.symbiotic.network)
+                    .call()
+                    .await?
+                    ._0
+                    {
+                        eyre::bail!(
+                            "Operator with address {} not opted in to the bolt network ({})",
+                            signer.address(),
+                            deployments.symbiotic.network
+                        );
+                    }
+
+                    let middleware = BoltSymbioticMiddleware::new(
+                        deployments.bolt.symbiotic_middleware,
+                        provider.clone(),
+                    );
+
+                    let pending =
+                        middleware.registerOperator(operator_rpc.to_string()).send().await?;
+
+                    info!(
+                        hash = ?pending.tx_hash(),
+                        "registerOperator transaction sent, awaiting receipt..."
+                    );
+
+                    let receipt = pending.get_receipt().await?;
+                    if !receipt.status() {
+                        eyre::bail!("Transaction failed: {:?}", receipt)
+                    }
+
+                    info!("Succesfully registered Symbiotic operator");
+
+                    Ok(())
+                }
+                SymbioticSubcommand::Status { rpc_url, address } => {
+                    let provider = ProviderBuilder::new().on_http(rpc_url.clone());
+                    let chain_id = provider.get_chain_id().await?;
+                    let chain = Chain::from_id(chain_id)
+                        .unwrap_or_else(|| panic!("chain id {} not supported", chain_id));
+
+                    let deployments = deployments_for_chain(chain);
+                    let bolt_manager =
+                        BoltManagerContract::new(deployments.bolt.manager, provider.clone());
+                    if bolt_manager.isOperator(address).call().await?._0 {
+                        info!(?address, "Symbiotic operator is registered");
+                    } else {
+                        warn!(?address, "Operator not registered");
+                    }
+
+                    Ok(())
                 }
             },
         }
     }
 }
 
+fn request_confirmation() -> bool {
```

Comment:
we can use inquire for this kind of thing. https://docs.rs/inquire/latest/inquire/struct.Confirm.html

Reply:
It's cool but considering current usage I'm okay with having a little function and not having another dependency

Reply:
Sounds good, however I'm OK adding this dependency down the line. It also enables interactivity in the CLI.